providers/mlx5/verbs.c

```diff
@@ -4572,6 +4664,16 @@ void clean_dyn_uars(struct ibv_context *context)
 		mlx5_free_uar(context, bf);
 	}
 
+	list_for_each_safe(&ctx->dyn_uar_qp_dedicated_list, bf, tmp_bf, uar_entry) {
+		list_del(&bf->uar_entry);
+		mlx5_free_uar(context, bf);
+	}
+
+	list_for_each_safe(&ctx->dyn_uar_qp_shared_list, bf, tmp_bf, uar_entry) {
+		list_del(&bf->uar_entry);
+		mlx5_free_uar(context, bf);
+	}
+
 	if (ctx->cq_uar)
```

Comment:
cq doesn't get dynamic too?

Reply:
It gets when kernel supports the new mode, otherwise the static one for CQs will be used.